src/timer/timer_game_realtime.h

```diff
@@ -0,0 +1,59 @@
+/*
+ * This file is part of OpenTTD.
+ * OpenTTD is free software; you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, version 2.
+ * OpenTTD is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details. You should have received a copy of the GNU General Public License along with OpenTTD. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/** @file timer_game_realtime.h Definition of the real time game-timer */
+
+#ifndef TIMER_GAME_REALTIME_H
+#define TIMER_GAME_REALTIME_H
+
+#include <chrono>
+
+/**
+ * Timer that represents real time for game-related purposes.
+ *
+ * For pausing, there are several modes:
+ * - Continue to tick during pause (PeriodFlags::ALWAYS).
+ * - Stop ticking when paused (PeriodFlags::UNPAUSED).
+ * - Only tick when unpaused or when there was a Command executed recently (recently: since last autosave) (PeriodFlags::AUTOSAVE).
+ *
+ * @note The lowest possible interval is 1ms, although realistic the lowest
+ * interval is 27ms. This timer is only updated when the game-thread makes
+ * a tick, which happens every 27ms.
+ * @note Callbacks are executed in the game-thread.
+ */
+class TimerGameRealtime {
+public:
+	enum PeriodFlags {
```

Comment:
I know this is scoped already, but enum class?

Reply:
I kinda deliberately made them non-class-enums, so you can do `TimerGameRealtime::AUTOSAVE` instead of `TimerGameRealtime::PeriodFlags::AUTOSAVE`, to make the API slightly easier.

Similar thing I did for the other timers.

If we don't like that, I can change it, but it is a lot more letters for no real good reason in my opinion :) Let me know what you think!